### DISCUSSION TITLE
Update 5.3.6 to 5.4.1 - test failed

### DISCUSSION THREAD

**Discussion by coogor** [miguelgrinberg/Flask-SocketIO#2111] [Q&A]
Hello,
I'm trying to upgrade from 5.3.6 to 5.4.1 for openSUSE Tumbleweed (Rolling Release) and run into a test error:

[code]
```
[    9s] + python3.10 -m unittest -v test_socketio.py
[   10s] Traceback (most recent call last):
[   10s]   File "/usr/lib64/python3.10/runpy.py", line 196, in _run_module_as_main
[   10s]     return _run_code(code, main_globals, None,
[   10s]   File "/usr/lib64/python3.10/runpy.py", line 86, in _run_code
[   10s]     exec(code, run_globals)
[   10s]   File "/usr/lib64/python3.10/unittest/__main__.py", line 18, in <module>
[   10s]     main(module=None)
[   10s]   File "/usr/lib64/python3.10/unittest/main.py", line 100, in __init__
[   10s]     self.parseArgs(argv)
[   10s]   File "/usr/lib64/python3.10/unittest/main.py", line 147, in parseArgs
[   10s]     self.createTests()
[   10s]   File "/usr/lib64/python3.10/unittest/main.py", line 158, in createTests
[   10s]     self.test = self.testLoader.loadTestsFromNames(self.testNames,
[   10s]   File "/usr/lib64/python3.10/unittest/loader.py", line 220, in loadTestsFromNames
[   10s]     suites = [self.loadTestsFromName(name, module) for name in names]
[   10s]   File "/usr/lib64/python3.10/unittest/loader.py", line 220, in <listcomp>
[   10s]     suites = [self.loadTestsFromName(name, module) for name in names]
[   10s]   File "/usr/lib64/python3.10/unittest/loader.py", line 154, in loadTestsFromName
[   10s]     module = __import__(module_name)
[   10s]   File "/home/abuild/rpmbuild/BUILD/flask_socketio-5.4.1/test_socketio.py", line 11, in <module>
[   10s]     socketio = SocketIO(app)
[   10s]   File "/home/abuild/rpmbuild/BUILDROOT/python-Flask-SocketIO-5.4.1-0.x86_64/usr/lib/python3.10/site-packages/flask_socketio/__init__.py", line 187, in __init__
[   10s]     self.init_app(app, **kwargs)
[   10s]   File "/home/abuild/rpmbuild/BUILDROOT/python-Flask-SocketIO-5.4.1-0.x86_64/usr/lib/python3.10/site-packages/flask_socketio/__init__.py", line 243, in init_app
[   10s]     self.server = socketio.Server(**self.server_options)
[   10s]   File "/usr/lib/python3.10/site-packages/socketio/base_server.py", line 31, in __init__
[   10s]     self.eio = self._engineio_server_class()(**engineio_options)
[   10s]   File "/usr/lib/python3.10/site-packages/engineio/base_server.py", line 81, in __init__
[   10s]     raise ValueError('Invalid async_mode specified')
[   10s] ValueError: Invalid async_mode specified
[   10s] error: Bad exit status from /var/tmp/rpm-tmp.vl0P2D (%check)
```
[/code]

Any idea?
Thanks

**Comment by miguelgrinberg** [comment]
I don't know what to suggest, as the tests pass on my CI. My guess is that this isn't related to the update but to some other condition in your environment, like maybe the dependencies that you have installed or any changes or patches you may have applied to the code.

**Comment by coogor** [comment]
On what system are you working?
My best guess is that one dependency got updated to a newer release, which is obviously incompatible. 
On a rolling release one sees these issues quite quickly. Since one week the current 5.3.6 version fails in Tumbleweed as well. I'm currently trying to find out which new python-packages have arrived by then

**Comment by danigm** [comment]
I've found that the problem was in a missing dependency for package python-python-engineio. I've just fixed the issue there, so it should build as soon as this update lands Tumbleweed.

https://build.opensuse.org/request/show/1222660

**Reply by daerlang** [reply]
i still faced the issue Invalid async_mode specified. need some update?
run my app in dev env is ok but pyinstaller to exe to run is error as below: 
my pip list as bellow:
`Traceback (most recent call last):
  File "app\vncview.py", line 578, in <module>
  File "flask_socketio\__init__.py", line 187, in __init__
  File "flask_socketio\__init__.py", line 243, in init_app
  File "socketio\base_server.py", line 31, in __init__
  File "engineio\base_server.py", line 81, in __init__
ValueError: Invalid async_mode specified   ` 

`(vncview_env) PS D:\python> pip list
Package                   Version
------------------------- ---------
aiohappyeyeballs          2.4.4    
aiohttp                   3.11.9   
aiosignal                 1.3.1    
altgraph                  0.17.4   
async-timeout             4.0.3    
attrs                     24.2.0   
bidict                    0.23.1   
blinker                   1.9.0    
certifi                   2024.8.30
cffi                      1.17.1
charset-normalizer        3.4.0
click                     8.1.7
colorama                  0.4.6
cryptography              44.0.0
dnspython                 2.7.0
eventlet                  0.38.1
Flask                     3.1.0
Flask-SocketIO            5.4.1
frozenlist                1.5.0
gevent                    24.11.1
gevent-websocket          0.10.1
greenlet                  3.1.1
h11                       0.14.0
idna                      3.10
itsdangerous              2.2.0
Jinja2                    3.1.4
JsonForm                  0.0.2
jsonschema                4.23.0
jsonschema-specifications 2024.10.1
JsonSir                   0.0.2
jwcrypto                  1.5.6
MarkupSafe                3.0.2
multidict                 6.1.0
mysql-connector-python    9.1.0
numpy                     2.1.3
nvidia-ml-py              12.560.30
packaging                 24.2
pefile                    2023.2.7
pillow                    11.0.0
pip                       24.3.1
propcache                 0.2.1
psutil                    6.1.0
py3nvml                   0.2.7
pycparser                 2.22
pyinstaller               6.11.1
pyinstaller-hooks-contrib 2024.10
pyngrok                   7.2.1
pynvml                    12.0.0
Python-EasyConfig         0.1.7
python-engineio           4.10.1
python-socketio           5.11.4
pywin32                   308
pywin32-ctypes            0.2.3
PyYAML                    6.0.2
redis                     5.2.0
referencing               0.35.1
requests                  2.32.3
Resource                  0.2.1
rpds-py                   0.22.1
setuptools                65.5.0
simple-websocket          1.1.0
six                       1.16.0
urllib3                   2.2.3
wakeonlan                 3.1.0
Werkzeug                  3.1.3
WMI                       1.5.1
wsproto                   1.2.0
xmltodict                 0.14.2
yarl                      1.18.3
zope.event                5.0
zope.interface            7.2`

pyinstaller include 
```
25275 DEBUG: Hidden import 'engineio' already found
25275 DEBUG: Hidden import 'simple_websocket' already found`
```

my flask code:
`app = Flask(__name__, static_folder=r'D:\python\vncview\app\noVNC-1.5.0', static_url_path='/')
socketio = SocketIO(app)`

**Comment by coogor** [comment]
Am Freitag, 8. November 2024, 08:53:18 Mitteleuropäische Normalzeit schrieb danigm:
> 
> I&#39;ve found that the problem was in a missing dependency for package python-python-engineio. I&#39;ve just fixed the issue there, so it should build as soon as this update lands Tumbleweed.
> 
> https://build.opensuse.org/request/show/1222660
> 
> 
Thank you!
